node/network/bitfield-distribution/src/lib.rs

```diff
@@ -344,18 +371,34 @@ async fn relay_message<Context>(
 	.await;
 
 	drop(_span);
+	let total_peers = peer_views.len();
+	let mut random_routing: RandomRouting = Default::default();
 
 	let _span = span.child("interested-peers");
 	// pass on the bitfield distribution to all interested peers
-	let mut interested_peers = peer_views
+	let interested_peers = peer_views
 		.iter()
 		.filter_map(|(peer, view)| {
 			// check interest in the peer in this message's relay parent
 			if view.contains(&message.relay_parent) {
 				let message_needed =
 					job_data.message_from_validator_needed_by_peer(&peer, &validator);
 				if message_needed {
-					Some(peer.clone())
+					let in_topology = topology.route_to_peer(required_routing, &peer);
+					let need_routing = in_topology || {
+						let route_random = random_routing.sample(total_peers, rng);
+						if route_random {
+							random_routing.inc_sent();
+						}
+
+						route_random
+					};
+
+					if need_routing {
+						Some(peer.clone())
+					} else {
+						None
+					}
```

Comment:
Not sure about the usage of `RandomRouting`. This might send to up to 25 more random peers. We can maybe try to process our gossip peers first, then this will reduce the potential amount of random peers.
But the same is true for approval-distribution I think.

Reply:
Yes, I was also thinking about it. I'm currently trying to sketch up a unit test to check this part.

Reply:
Actually the default target for random circulation is `DEFAULT_RANDOM_CIRCULATION = 4` so we actually send the message for all targeted gossip peers + up to 4 random peers.

Reply:
You're right. If we always want to add up to 4 random peers in the mix, then it should be fine.